pkgs/by-name/nv/nvidia-container-toolkit/package.nix

```diff
@@ -56,6 +39,12 @@ buildGoModule rec {
     ./0001-Add-dlopen-discoverer.patch
   ];
 
+  subPackages = [
+    "cmd/nvidia-ctk"
+    "cmd/nvidia-container-runtime-hook"
+    "cmd/nvidia-container-runtime.legacy"
+  ];
```

Comment:
I'm still not convinced why we should filter the targets for the top-level attribute; ofc it's not such a big deal since the user still can `.overrideAttrs (_: { subpackages = [ ... ]; })`, but I don't understand the motivation

Reply:
https://github.com/NVIDIA/nvidia-container-toolkit/tree/a470818ba7d9166be282cd0039dd2fc9b0a34d73/tools/container are just tools to configure a host in an imperative way.

Reply:
Sooo why do we take care to exclude them if upstream decided to put them in the same package?

Reply:
Do we want to provide applications that are exclusively for configuring the host? We can add them on a separate output as we have discussed, but are they really of any use in our case? I would say no, willing to add them when we get some real use case in case that happens.

Reply:
> We can add them on a separate output as we have discussed, but are they really of any use in our case? I would say no, willing to add them when we get some real use case in case that happens.

Our NixOS module is one thing, the nixpkgs package is another. The packages usually reflect whatever decisions upstream projects make. We deviate to make things more deterministic and unbreak them on nixos

Reply:
Okay, going to update the PR.

Reply:
Sorry, have been out. Updated.